# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ConcurrentDelazification.h
# Commit: 31d3f2901db9
# Full Hash: 31d3f2901db927a29898c5208a2624b4225382c1
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-12-05 09:58:04
# Regressor Bug: 1933681
# File Overlap Count: 1
# Description:
#   Bug 1933681 - Part 1: Allocate InitialStencilAndDelazifications in CompileGlobalScriptToStencilAndMaybeInstantiate. r=nbp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D230387
# ==============================================================================

diff -r 2694b90c0deb -r 31d3f2901db9 js/src/vm/ConcurrentDelazification.h
--- a/js/src/vm/ConcurrentDelazification.h	Thu Dec 05 02:39:42 2024 +0200
+++ b/js/src/vm/ConcurrentDelazification.h	Wed Dec 04 23:44:33 2024 +0000
@@ -12,9 +12,9 @@
 #include <stddef.h>  // size_t
 #include <utility>   // std::pair
 
-#include "frontend/CompilationStencil.h"  // frontend::{CompilationStencil, ScriptStencilRef, CompilationStencilMerger}
-#include "frontend/ScriptIndex.h"  // frontend::ScriptIndex
-#include "js/AllocPolicy.h"        // SystemAllocPolicy
+#include "frontend/CompilationStencil.h"  // frontend::{InitialStencilAndDelazifications, CompilationStencil, ScriptStencilRef, CompilationStencilMerger}
+#include "frontend/ScriptIndex.h"         // frontend::ScriptIndex
+#include "js/AllocPolicy.h"               // SystemAllocPolicy
 #include "js/CompileOptions.h"  // JS::PrefableCompileOptions, JS::ReadOnlyCompileOptions
 #include "js/UniquePtr.h"  // UniquePtr
 #include "js/Vector.h"     // Vector
@@ -125,7 +125,7 @@
         stackQuota_(stackQuota) {}
 
   bool init(const JS::ReadOnlyCompileOptions& options,
-            const frontend::CompilationStencil& stencil);
+            frontend::InitialStencilAndDelazifications* stencils);
   bool delazify();
 
   // This function is called by `delazify` function to know whether the